packages/image_picker/image_picker_platform_interface/lib/src/platform_interface/image_picker_platform.dart

```diff
@@ -210,6 +210,81 @@ abstract class ImagePickerPlatform extends PlatformInterface {
     throw UnimplementedError('getMultiImage() has not been implemented.');
   }
 
+  /// Returns an [XFile] with the image or video that was picked from the gallery.
+  ///
+  /// The images come from the [ImageSource.gallery].
+  ///
+  /// Where iOS supports HEIC images, Android 8 and below doesn't. Android 9 and
+  /// above only support HEIC images if used in addition to a size modification,
+  /// of which the usage is explained below.
+  ///
+  /// This method is only supported on Android API 19 and above.
+  /// It could possibly work on API 18 and below on some devices,
+  /// but this is not guaranteed.
+  ///
+  /// For when an image is picked:
+  ///
+  ///   If specified, the image will be at most [maxImageWidth] wide and
+  ///   [maxImageHeight] tall. Otherwise the image will be returned at its
+  ///   original width and height.
+  ///
+  ///   The [imageQuality] argument modifies the quality of the image, ranging
+  ///   from 0-100 where 100 is the original/max quality. If `imageQuality` is
+  ///   null, the image with the original quality will be returned. Compression
+  ///   is only supported for certain image types such as JPEG and on Android
+  ///   PNG and WebP. If compression is not supported for the image that is
+  ///   picked, a warning message will be logged.
+  ///
+  /// In Android, the MainActivity can be destroyed for various reasons. If that happens, the result will be lost
+  /// in this call. You can then call [getLostData] when your app relaunches to retrieve the lost data.
+  ///
+  /// If no image or video were picked, the return value is null.
+  Future<XFile?> getImageOrVideo({
```

Comment:
Maybe we should call this `getGalleryImageOrVideo`, to make it more clear why this is a totally separate method instead of a replacement for `getImage` with an extra parameter for allowing video (which I was going to suggest until I realized it doesn't support `source`).

Reply:
Although... is getting image or video but with the camera source something we could support in the future? If so, we should probably just make a replacement method.